### PR TITLE
962 modal optional sizes

### PR DESCRIPTION
**Pull request by dexguitar**

### CODE REVIEW

File: jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/modal/ModalVerticallyCenteredTests.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+    @Test(dataProvider = "modalVerticalAlignmentData")
+    public void modalVerticalAlignmentTest(Button showButton,
+                                           Button dismissButton,
+                                           String waiterSelector,
+                                           String modalId) {
+        showButton.show();
+        showButton.click();
+
+        $(waiterSelector).waitFor().hidden();
```
Comment (AUlasen) [inline jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/modal/ModalVerticallyCenteredTests.java]: You should use elements, not raw methods and locators.

File: jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/modal/ModalVerticallyCenteredTests.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+        long modalTop = WebDriverFactory.jsExecute(
+                "var modal = document.getElementById('" + modalId + "');" +
+                        "return modal.getBoundingClientRect().top;"
+        );
+        long modalBottom = WebDriverFactory.jsExecute(
+                "var modal = document.getElementById('" + modalId + "');" +
+                        "return window.innerHeight - modal.getBoundingClientRect().bottom;"
```
Comment (AUlasen) [inline jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/modal/ModalVerticallyCenteredTests.java]: The same as one of the  above. Method `core().getRect() `may be useful.

File: jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/modal/ModalOptionalSizesTests.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                {modalOptionalSizes.smButton, modalOptionalSizes.smModal, "modal-sm"},
+        };
+    }
+
+    @DataProvider
+    public Object[][] modalSizeData() {
+        return new Object[][]{
+                {modalOptionalSizes.xlButton, modalOptionalSizes.xlModal, "bd-example-modal-xl", 1140},
+                {modalOptionalSizes.lgButton, modalOptionalSizes.lgModal, "bd-example-modal-lg", 800},
+                {modalOptionalSizes.smButton, modalOptionalSizes.smModal, "bd-example-modal-sm", 300},
```
[the remark this answers is not in the document]
Comment (dexguitar) [reply jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/modal/ModalOptionalSizesTests.java]: Refactored. Now using getRect() there.

File: jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/modal/ModalOptionalSizesTests.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                              String modalClassName,
+                              long modalSize) {
+        button.highlight();
+        button.click();
+
+        modal.is().displayed();
+
+        long modalWidth = WebDriverFactory.jsExecute(
+                "var modal = document.getElementsByClassName('" + modalClassName + "')[0];" +
+                        "return modal.children[0].offsetWidth;"
```
[the remark this answers is not in the document]
Comment (dexguitar) [reply jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/modal/ModalOptionalSizesTests.java]: Refactored. Now using getRect() there.

**Pull request merged**